### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by akopp90**

### CODE REVIEW

File: Models/User.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    image: String,
+    emailVerified: Date,
+  },
+  {
+    timestamps: true,
+  }
+);
+
+userSchema.pre("save", async function (next) {
+  if (!this.isModified("password")) return next();
```
Comment (F-Kirchhoff) [inline Models/User.js:37]:
```suggestion
  if (!this.isModified("password")) {
    next();
    return;
  };
```

File: components/layout/Login.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import Link from "next/link";
+export default function Login() {
+  const { data: session } = useSession();
+  const router = useRouter();
+  const [credentials, setCredentials] = useState({
+    email: "",
+    password: "",
+  });
+
+  const handleSubmit = async (e) => {
```
Comment (F-Kirchhoff) [inline components/layout/Login.js:17]: pleave don't use abbreviations like e. Also watch out for code consistency, so please use the function syntax for these functions.

File: components/layout/Login.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  return (
+    <Container>
+      <FormCard>
+        <Title>Sign In</Title>
+        <form onSubmit={handleSubmit}>
+          <FormGroup>
+            <Input
+              name={"email"}
+              type="email"
+              onChange={(e) =>
```
Comment (F-Kirchhoff) [inline components/layout/Login.js:46]: here as well, please use `event`

File: components/layout/Login.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+              }
+            >
+              Email
+            </Input>
+          </FormGroup>
+          <FormGroup>
+            <Input
+              type="password"
+              name="password"
+              onChange={(e) =>
```
Comment (F-Kirchhoff) [inline components/layout/Login.js:57]: and here.

File: components/layout/RegisterForm.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+function RegisterForm() {
+  const router = useRouter();
+  const [userData, setUserData] = useState({
+    name: "",
+    email: "",
+    password: "",
+  });
+
+  const handleSubmit = async (e) => {
```
Comment (F-Kirchhoff) [inline components/layout/RegisterForm.js:41]: please check your code consistency across your entire PR, regarding abbreviations and syntax.

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey Activity Team,
very advanced PR, looks good so far. There is only a problem regarding the code consistency. Please make sure to write code that is easy to read, not as short or new as possible. Especially abbreviations and different ways of creating the same thing make it harder for others to digest the code quickly. Please adapt this code accordingly.

**Review by Bianca-Gundel — APPROVED** [review]

**Review by F-Kirchhoff — APPROVED** [review]

**Pull request merged**